datashader/core.py

```diff
@@ -167,6 +170,61 @@ def line(self, source, x, y, agg=None):
             agg = any()
         return bypixel(source, self, Line(x, y), agg)
 
+    def raster(self,
+               source,
+               band=1,
+               nodata=np.nan,
+               replace_nodata=None):
+        """Sample a raster dataset by canvas size and bounds. Note: requires
+        `rasterio` and `scikit-image`.
+
+        Parameters
+        ----------
+        source : rasterio.Dataset
+            input datasource most likely obtain from `rasterio.open()`.
+        band : int
+            source band number : optional default=1
+        nodata : number
+            Value for nodata cells
+        replace_nodata : number
+            Replacement value for nodata cells
```

Comment:
I wonder if there's a better api for these keywords. Also, much of the transfer-function stuff (for better or worse) assumes that NaN is missing for floats, and 0 otherwise. One thought would be to have:

```rst
missing : number
    Missing flag, default is `NaN`. Missing values are replaced with `NaN`
    if floats, and 0 otherwise.
```

If the user wants more control they could handle this themselves using Xarray after the load (or this could be entirely removed in favor of that):

```python
agg = cvs.raster(source)
agg2 = agg.where(agg == 0)    # NaN where agg is 0
# or
agg3 = agg.fillna(0)          # 0 where agg is NaN
```